substrate/client/network/src/protocol/notifications/handler.rs

```diff
@@ -935,679 +931,682 @@ pub mod tests {
-		));
-
-		// inject dial failure to an already closed substream and verify outbound state is reset
-		handler.on_connection_event(handler::ConnectionEvent::DialUpgradeError(
-			handler::DialUpgradeError { info: 0, error: ConnectionHandlerUpgrErr::Timeout },
-		));
-		assert!(std::matches!(
-			handler.protocols[0].state,
-			State::OpenDesiredByRemote { pending_opening: false, .. }
-		));
-	}
-
-	#[tokio::test]
-	async fn sync_notifications_clogged() {
-		let mut handler = notifs_handler();
-		let (io, _) = MockSubstream::negotiated().await;
-		let codec = UviBytes::default();
-
-		let (async_tx, async_rx) = futures::channel::mpsc::channel(ASYNC_NOTIFICATIONS_BUFFER_SIZE);
-		let (sync_tx, sync_rx) = futures::channel::mpsc::channel(1);
-		let notifications_sink = NotificationsSink {
-			inner: Arc::new(NotificationsSinkInner {
-				peer_id: PeerId::random(),
-				async_channel: FuturesMutex::new(async_tx),
-				sync_channel: Mutex::new(Some(sync_tx)),
-			}),
-		};
-
-		handler.protocols[0].state = State::Open {
-			notifications_sink_rx: stream::select(async_rx.fuse(), sync_rx.fuse()).peekable(),
-			out_substream: Some(NotificationsOutSubstream::new(Framed::new(io, codec))),
-			in_substream: None,
-		};
-
-		notifications_sink.send_sync_notification(vec![1, 3, 3, 7]);
-		notifications_sink.send_sync_notification(vec![1, 3, 3, 8]);
-		notifications_sink.send_sync_notification(vec![1, 3, 3, 9]);
-		notifications_sink.send_sync_notification(vec![1, 3, 4, 0]);
-
-		futures::future::poll_fn(|cx| {
-			assert!(std::matches!(
-				handler.poll(cx),
-				Poll::Ready(ConnectionHandlerEvent::Close(
-					NotifsHandlerError::SyncNotificationsClogged,
-				))
-			));
-			Poll::Ready(())
-		})
-		.await;
-	}
-
-	#[tokio::test]
-	async fn close_desired_by_remote() {
-		let mut handler = notifs_handler();
-		let (io, io2) = MockSubstream::negotiated().await;
-		let mut codec = UviBytes::default();
-		codec.set_max_len(usize::MAX);
-
-		let notif_in = NotificationsInOpen {
-			handshake: b"hello, world".to_vec(),
-			negotiated_fallback: None,
-			substream: NotificationsInSubstream::new(
-				Framed::new(io, codec),
-				NotificationsInSubstreamHandshake::PendingSend(vec![1, 2, 3, 4]),
-			),
-		};
-
-		// add new inbound substream but close it immediately and verify that correct events are
-		// emitted
-		handler.on_connection_event(handler::ConnectionEvent::FullyNegotiatedInbound(
-			handler::FullyNegotiatedInbound { protocol: (notif_in, 0), info: () },
-		));
-		drop(io2);
-
-		futures::future::poll_fn(|cx| {
-			assert!(std::matches!(
-				handler.poll(cx),
-				Poll::Ready(ConnectionHandlerEvent::Custom(
-					NotifsHandlerOut::OpenDesiredByRemote { protocol_index: 0 },
-				))
-			));
-			assert!(std::matches!(
-				handler.poll(cx),
-				Poll::Ready(ConnectionHandlerEvent::Custom(NotifsHandlerOut::CloseDesired {
-					protocol_index: 0
-				},))
-			));
-			Poll::Ready(())
-		})
-		.await;
 	}
 }
+
+// 	}
```

Comment:
Dq: This should be enabled before merging or as a followup? Maybe an issue with a todo comment in the latter case